org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codeactions/CodeActionMarkerResolution.java

```diff
@@ -92,8 +95,16 @@ public void run(IMarker marker) {
 					Command command = codeAction.getCommand();
 					ExecuteCommandOptions provider = wrapper.getServerCapabilities().getExecuteCommandProvider();
 					if (provider != null && provider.getCommands().contains(command.getCommand())) {
+						final String serverId = wrapper.serverDefinition.id;
 						wrapper.execute(ls -> ls.getWorkspaceService()
-								.executeCommand(new ExecuteCommandParams(command.getCommand(), command.getArguments())));
+								.executeCommand(new ExecuteCommandParams(command.getCommand(), command.getArguments()))).exceptionally(t -> {
+									String errorTitle = "Error Executing Quick Fix"; //$NON-NLS-1$
+									String errorMessage = "Failed to fetch quick fix edit for '" + codeAction.getTitle() //$NON-NLS-1$
+											+ "'. See Language Server '" + serverId + "' log for more details."; //$NON-NLS-1$ //$NON-NLS-2$
+									Display display = PlatformUI.getWorkbench().getDisplay();
+									display.asyncExec(() -> MessageDialog.openError(display.getActiveShell(), errorTitle, errorMessage));
```

Comment:
I think we should rather use ServerMessageHandler.showMessage, as in other places in the codebase. Could you do that?